pkg/receive/handler.go

```diff
@@ -71,3 +92,33 @@ func (h *Handler) Receive(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 }
+
+var ErrRequiredLabelMissing = fmt.Errorf("a required label is missing from the metric")
+
+// TODO: Make this a middleware eventually
```

Comment:
can we actually make this part of this same PR? Now that we are rewriting key pieces of telemetry, it's a good time to start writing more maintainable and composable funcs. Let's make a write-request validation middleware that can take a variable amount of `func(* prompb.WriteRequest) error` funcs and returns a handler. This way we can wrap the forwarder easily and separate concerns

Reply:
can we still address this ^ before merging?